plugins/SpectrumAnalyzer/SaControlsDialog.cpp

```diff
@@ -0,0 +1,227 @@
+	config_layout->addWidget(refFreezeButton, 2, 0, 2, 1);
+
+	// misc configuration switches
+	LedCheckBox *waterfallButton = new LedCheckBox(tr("Waterfall"), this);
+	waterfallButton->setToolTip(tr("Display real-time spectrogram"));
+	waterfallButton->setCheckable(true);
+	waterfallButton->setMinimumSize(70, 12);
+	waterfallButton->setModel(&controls->m_waterfallModel);
+	config_layout->addWidget(waterfallButton, 0, 1);
+
+	LedCheckBox *smoothButton = new LedCheckBox(tr("Averaging"), this);
+	smoothButton->setToolTip(tr("Enable exponential moving average"));
+	smoothButton->setCheckable(true);
+	smoothButton->setMinimumSize(70, 12);
+	smoothButton->setModel(&controls->m_smoothModel);
+	config_layout->addWidget(smoothButton, 1, 1);
+
+	LedCheckBox *stereoButton = new LedCheckBox(tr("Stereo"), this);
+	stereoButton->setToolTip(tr("Display stereo channels separately"));
+	stereoButton->setCheckable(true);
+	stereoButton->setMinimumSize(70, 12);
+	stereoButton->setModel(&controls->m_stereoModel);
+	config_layout->addWidget(stereoButton, 2, 1);
+
+	LedCheckBox *peakHoldButton = new LedCheckBox(tr("Peak hold"), this);
+	peakHoldButton->setToolTip(tr("Display envelope of peak values"));
+	peakHoldButton->setCheckable(true);
+	peakHoldButton->setMinimumSize(70, 12);
+	peakHoldButton->setModel(&controls->m_peakHoldModel);
+	config_layout->addWidget(peakHoldButton, 3, 1);
+
+	// frequency: linear / log. switch and range selector
+	PixmapButton *logXButton = new PixmapButton(this, tr("Logarithmic frequency"));
+	logXButton->setToolTip(tr("Switch between logarithmic and linear frequency scale"));
+	QPixmap *logXOnPixmap = new QPixmap(PLUGIN_NAME::getIconPixmap("x_log").scaled(iconSize, iconSize, Qt::IgnoreAspectRatio, Qt::SmoothTransformation));
+	QPixmap *logXOffPixmap = new QPixmap(PLUGIN_NAME::getIconPixmap("x_linear").scaled(iconSize, iconSize, Qt::IgnoreAspectRatio, Qt::SmoothTransformation));
+	logXOnPixmap->setDevicePixelRatio(devicePixelRatio());
+	logXOffPixmap->setDevicePixelRatio(devicePixelRatio());
+	logXButton->setActiveGraphic(*logXOnPixmap);
+	logXButton->setInactiveGraphic(*logXOffPixmap);
+	logXButton->setCheckable(true);
+	logXButton->setModel(&controls->m_logXModel);
+	config_layout->addWidget(logXButton, 0, 2, 2, 1, Qt::AlignRight);
+
+	ComboBox *freqRangeCombo = new ComboBox(this, tr("Frequency range"));
+	freqRangeCombo->setToolTip(tr("Frequency range"));
+	freqRangeCombo->setMinimumSize(100, 22);
+	freqRangeCombo->setMaximumSize(200, 22);
+	freqRangeCombo->setModel(&controls->m_freqRangeModel);
+	config_layout->addWidget(freqRangeCombo, 0, 3, 2, 1);
+
+	// amplitude: linear / log switch and range selector
+	PixmapButton *logYButton = new PixmapButton(this, tr("Logarithmic amplitude"));
+	logYButton->setToolTip(tr("Switch between logarithmic and linear amplitude scale"));
+	QPixmap *logYOnPixmap = new QPixmap(PLUGIN_NAME::getIconPixmap("y_log").scaled(iconSize, iconSize, Qt::IgnoreAspectRatio, Qt::SmoothTransformation));
```

Comment:
That line is too long.

Reply:
You will find more long lines throughout the code; I actually started splitting them before making the PR, but after doing a file or two, I had to go back and undo most of the changes, because the code became almost unreadable in some places.
In this case, specifically, there is not really any good place to put a line break, in my opinion. So I prefer leaving it as it is -- the text editor will wrap the line in a way that makes it obvious it is one logical piece.
If I insert a line break, it makes the readability worse, because you actually have to go through both lines and parse what's happening (mainly on the second line), whereas ordinarily you would just see "x  = new Something(..." and move on if it wasn't what you were looking for.
So in the end I decided not to force long lines into compliance at all cost, and in general I only split lines that are over 120 characters and offer some reasonable place to insert the line break (usually between function parameters).

Reply:
Even on my large screen at home, that line is covered at 120 and I can't see more (I think it's 140). So 120 should always be the maximum. In this case though, the end of the line is not that interesting, maybe it's better you are hiding it... :smile: